subsys/mgmt/updatehub/updatehub.c

```diff
@@ -952,6 +983,19 @@ static void autohandler(struct k_work *work)
 
 void updatehub_autohandler(void)
 {
+	BUILD_ASSERT(
+		IS_ENABLED(CONFIG_UPDATEHUB_DOWNLOAD_SHA256_VERIFICATION) ||
+		IS_ENABLED(CONFIG_UPDATEHUB_STORAGE_SHA256_VERIFICATION),
+		"UpdateHub requires at least one SHA-256 verification "
+		"enabled and strictly recommends both as enabled");
```

Comment:
If you are so strict about this you may want to change the Kconfig configuration to `choice/endchoice` menu.
This way you may restrict users, in Kconfig menu, to select only from given combinations, eg:
```
( )SHA on download
( )SHA from flash
( )flash and download verification
```
You may check `zephyr/Kconfig.zephyr` for example in `Compiler Options->Optimization Level`

Reply:
I also prefer the choice as it makes an invalid choice impossible.